### PR TITLE
add check to differentiate the version either enterprise or community

### PR DESCRIPTION
**Pull request by aman-alfresco**

### CODE REVIEW

**Review by agologan — CHANGES_REQUESTED** [review]
Current version just assumes `isContentServiceInstalled` as a prerequisite in order to return whether or not this is an enterprise installation.
This is unexpected and brittle, a better api would just add a fun along the lines of `isEnterpriseDistribution(endpoint: String)`

**Review by agologan — CHANGES_REQUESTED** [review]
This is better but the call code is now duplicated between the two functions, maybe extract it in a new function?
Also there is no caching involved between the two calls? Are you ok with doing the call twice?

File: auth/src/main/java/com/alfresco/auth/DiscoveryService.kt [hunk 1]
```diff
@@ -54,14 +54,41 @@ class DiscoveryService(
...
-
                 data?.isAtLeast(MIN_ACS_VERSION) ?: false
             } catch (e: Exception) {
                 false
             }
         }
     }
 
+    /**
+     * Check whether it's enterprise distribution or not on [endpoint].
```
[the remark this answers is not in the document]
Comment (aman-alfresco) [reply auth/src/main/java/com/alfresco/auth/DiscoveryService.kt]: done

**Comment by aman-alfresco** [comment]
> This is better but the call code is now duplicated between the two functions, maybe extract it in a new function?
> Also there is no caching involved between the two calls? Are you ok with doing the call twice?

yes. ok with that

**Pull request merged**